service/worker/workerdeployment/activities.go

```diff
@@ -58,6 +58,22 @@ func (a *Activities) SyncWorkerDeploymentVersion(ctx context.Context, args *depl
 	}, nil
 }
 
+func (a *Activities) IsVersionMissingTaskQueues(ctx context.Context, args *deploymentspb.IsVersionMissingTaskQueuesArgs) (*deploymentspb.IsVersionMissingTaskQueuesResult, error) {
+	res, err := a.deploymentClient.IsVersionMissingTaskQueues(
+		ctx,
+		a.namespace,
+		args.PrevCurrentVersion,
+		args.NewCurrentVersion,
+	)
+	if err != nil {
+		// todo (Shivam): do we return a non-retryable error here since we want to fail the operation if this check has failed.
```

Comment:
If the call to deploymentClient / matchingClient failed, it could be due to a network partition, which is retryable, we don't really know if it's retryable or not, so I think we want the Activity to retry a few times.

Ultimately, if the task queues are missing, the grpc error type we should return to the user is `serviceerror.NewFailedPrecondition` because the state of the system needs to change for this request to work.